src/components/Question.tsx

```diff
@@ -1,31 +1,35 @@
 import styled from 'styled-components';
 import { KanaType } from '../types/kana';
+import InputWithLabel from './InputWithLabel';
 
 type QuestionProps = {
   question: KanaType;
   answers: KanaType[];
-  onChange: (answer: KanaType) => void;
+  onSelectAnswer: (answer: KanaType) => void;
 };
 
 export default function Question({
   question,
   answers,
-  onChange,
+  onSelectAnswer,
 }: QuestionProps) {
+  function handleChange(event: React.ChangeEvent, answer: KanaType) {
+    onSelectAnswer(answer);
+  }
+
   return (
     <Fieldset>
       <Legend>{question.kana}</Legend>
       {answers.map(answer => (
-        <RadioButtonGroup key={answer.id}>
-          <Input
-            id={answer.id}
-            type="radio"
-            name={question.kana}
-            value={answer.kana}
-            onChange={() => onChange(answer)}
-          />
-          <Label htmlFor={answer.id}>{answer.pronunciation}</Label>
-        </RadioButtonGroup>
+        <InputWithLabel
+          key={answer.id}
+          labelText={answer.pronunciation}
+          id={answer.id}
+          type="radio"
+          name={question.kana}
+          value={answer.kana}
+          onChange={event => handleChange(event, answer)}
```

Comment:
Just a FYI. If you don't need the event, you don't have to pass it along:
`onChange={ () => handleChange(answer) }`

Reply:
Thank you! :)